airbyte/_processors/base.py

```diff
@@ -203,150 +187,44 @@ def process_airbyte_messages(
                 # Type.LOG, Type.TRACE, Type.CONTROL, etc.
                 pass
 
-        # We are at the end of the stream. Process whatever else is queued.
-        for stream_name, stream_batch in stream_batches.items():
-            batch_df = pd.DataFrame(stream_batch)
-            record_batch = pa.Table.from_pandas(batch_df)
-            self._process_batch(stream_name, record_batch)
-            progress.log_batch_written(stream_name, len(stream_batch))
-
-        all_streams = list(self._pending_batches.keys())
-        # Add empty streams to the streams list, so we create a destination table for it
-        for stream_name in self.expected_streams:
```

Comment:
are we still tackling empty streams downstream ?

Reply:
Great question - and I just double checked. Yes, we are still running regardless. Here's how it works now in the `SqlProcessorBase` class:

1. `write_all_stream_data()` is called at the end of `process_airbyte_messages()`.
2. `write_all_stream_data()` calls `write_stream_data()` on all values contained in `expected_streams`, so no streams are missed even data is not received for the stream in question.
3. `write_stream_data()` does its processing flow (including `_ensure_schema_exists()` even if there are no records/batches to process.